datumaro/plugins/data_formats/datumaro/exporter.py

```diff
@@ -64,68 +65,89 @@ def items(self):
+                image._path = osp.join(
+                    self._context._images_dir, item.subset, self._context._make_image_filename(item)
                 )
+                self._context._save_image(item, image.path)
 
-            item_desc["image"] = {
-                "path": path,
-            }
-            if item.media.has_size:  # avoid occasional loading
-                item_desc["image"]["size"] = item.media.size
+            yield
+            image._path = path
         elif isinstance(item.media, PointCloud):
             pcd = item.media_as(PointCloud)
             path = pcd.path
+
             if self._context._save_media:
-                path = self._context._make_pcd_filename(item)
-                self._context._save_point_cloud(
-                    item, osp.join(self._context._pcd_dir, item.subset, path)
+                # Temporarily update pcd path and save it.
+                pcd._path = osp.join(
+                    self._context._pcd_dir, item.subset, self._context._make_pcd_filename(item)
                 )
+                self._context._save_point_cloud(item, pcd.path)
+
+                # Temporarily update pcd related images paths and save them.
+                for i, img in enumerate(sorted(pcd.extra_images, key=lambda v: v.path)):
+                    img.__path = img.path
+                    img._path = osp.join(
+                        self._context._related_images_dir,
+                        item.subset,
+                        item.id,
+                        f"image_{i}{self._context._find_image_ext(img)}",
+                    )
 
-            item_desc["point_cloud"] = {"path": path}
+                    if img.has_data:
+                        img.save(img.path)
 
-            images = sorted(pcd.extra_images, key=lambda v: v.path)
+            yield
+            pcd._path = path
             if self._context._save_media:
-                related_images = []
-                for i, img in enumerate(images):
-                    ri_desc = {}
+                for img in pcd.extra_images:
+                    img._path = img.__path
+                    del img.__path
+        else:
+            raise NotImplementedError
 
-                    # Images can have completely the same names or don't
-                    # have them at all, so we just rename them
-                    ri_desc["path"] = f"image_{i}{self._context._find_image_ext(img)}"
+    def add_item(self, item: DatasetItem):
+        annotations = []
+        item_desc = {
+            "id": item.id,
+            "annotations": annotations,
+        }
 
-                    if img.has_data:
-                        img.save(
-                            osp.join(
-                                self._context._related_images_dir,
-                                item.subset,
-                                item.id,
-                                ri_desc["path"],
-                            )
-                        )
```

Comment:
We may not need to save the image if the image has data. Is this correct reason you removed?

Reply:
No all the things to save the media (images and point cloud) go into
```python
@contextmanager
    def save_media(self, item: DatasetItem):
```